### PR TITLE
Updated documentation

### PR DESCRIPTION
**Pull request by AlessandroA**
The update includes:
- minor fixes to ensure consistency with other uVisor docs
- removal of one section that is now obsolete

@meriac @iriark01

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,8 +1,8 @@
...
 
-This is a `Hello World!` yotta executable created to show some of the security features provided by the uVisor. The application hosts a secure box that is able to generate a secret at runtime. The secret is stored in the box context, so it is protected by uVisor against the rest of the code.
+This is a Hello World yotta executable created to show some of the security features provided by the uVisor. The application hosts a secure box that is able to generate a secret at runtime. The secret is stored in the box context, so it is protected by uVisor against the rest of the code.
 
-We propose here a challenge. For the security model to be effective, the box secret should not be in any way leaked. You can attempt whatever approach you prefer to try and read the secret. By default, the main event continuosuly tries to verify its challenge against the box secret.
+We propose here a challenge. For the security model to be effective, the box secret should not be in any way leaked. You can attempt whatever approach you prefer to try and read the secret. By default, the main event in this application continuosuly tries to verify its challenge against the box secret.
```
Comment (iriark01) [inline README.md]: This is a challenge. For the security model to be effective, the box secret must not be leaked in any way. You can attempt whatever approach you prefer to try and read the secret. By default, the main event in this application continuously tries to verify its challenge against the box secret.

File: README.md [hunk 2]
```diff
@@ -1,8 +1,8 @@
...
 
 We already provide one example of such approach. See the [Run](#run) section for more information.
```
Comment (iriark01) [inline README.md]: We already provide one example of such an approach

File: README.md [hunk 3]
```diff
@@ -56,21 +50,7 @@ On OS X [st-flash](https://github.com/texane/stlink) is available via `brew inst
 
 Hit the reset button after flashing to start program execution. The application will be running right after you reset the processor. When the blue (K64F) or green (STM32F4) LED blinks with a 1s period, it means that the program is simply comparing the password with the challenge, the comparison is not successful and it is ready to try again.
```
Comment (iriark01) [inline README.md]: The application starts running right

File: README.md [hunk 4]
```diff
@@ -56,21 +50,7 @@ On OS X [st-flash](https://github.com/texane/stlink) is available via `brew inst
...
-```bash
-yotta update -l
-```
-If the build process still fails, consider cleaning the build environment:
-
-```bash
-yotta clean
-```
-Then retry the build procedure.
+If you press the on-board push-button (`SW2` on the K64F, `USER` on the STM32F4) you will trigger an interrupt service routine that tries to read the secret from the (leaked) secure box context pointer. If the read is successful the same LED starts blinking faster (period of 200ms). If the uVisor captures the denied access, the system will halt and the red LED will blink (`PERMISSION_DENIED` error code). On the K64F do not use `SW3`, as it triggers a non-maskable intterrupt, which is currently unsupported and results in a halt with a `NOT_ALLOWED` error code.
```
Comment (iriark01) [inline README.md]: If you press the on-board push-button (`SW2` on the K64F, `USER` on the STM32F4), you trigger an interrupt service routine that tries to read the secret from the (leaked) secure box context pointer. If the read is successful the same LED starts blinking faster (a period of 200ms). If the uVisor captures the denied access, the system halts and the red LED blinks (`PERMISSION_DENIED` error code). On the K64F do not use `SW3`, as it triggers a non-maskable interrupt, which is currently unsupported and results in a halt with a `NOT_ALLOWED` error code.

**Comment by AlessandroA** [comment]
Thank you for your edits @iriark01 ! I applied them all and restored the `cd uvisor-helloworld` step that had been removed in the original PR. Merging.

**Pull request merged**